### PR TITLE
scylla_cluster: load_from_repository: set default timeouts

### PR DESCRIPTION
**Pull request by bhalevy**
Commit 4f619b75ab048366f3a0f7971450d00485e910bc (scylladb/scylla-ccm#492)
Improved the regular expression used in `scylla_extract_mode` to better
detect the scylla_mode given the package url.

However, when we install from a local path, like
`/jenkins/workspace/scylla-master/dtest-debug/scylla/build/debug/dist/tar/scylla-debug-unified-5.4.0~dev-0.20231013.055f0617064d.x86_64.tar.gz`,
The regular expression fails to detect the mode since it falsely start the match at `/scylla-master`.

This change restricts the match to the last path component.

As seen in https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-debug/258/artifact/logs-full.debug.041/dtest-gw0.log
```
07:20:43,371 790     errors                         ERROR    conftest.py         :225  | test_cluster_expansion_with_cdc[Single_cluster]: test failed:
...
>               self.wait_for_binary_interface(from_mark=from_mark, process=self._process_scylla, timeout=t)
...
self = <ccmlib.scylla_node.ScyllaNode object at 0x7fbd89790f10>
exprs = 'Starting listening for CQL clients', from_mark = 0, timeout = 420
```

When `cassandra_version` is passed to `ScyllaCluster.__init__`
we set `self.scylla_mode = None` and the default timeouts are not
adjusted.

`scylla_mode` is set correctly later on in `load_from_repository`.
Call `__set_default_timeouts` again to adjust the timeout based
on the valid `scylla_mode`.

### CODE REVIEW

**Review by fruch — APPROVED** [review]
LGTM

File: ccmlib/scylla_cluster.py [hunk 1]
```diff
@@ -98,8 +98,7 @@ def start_nodes(self, nodes=None, no_wait=False, verbose=False, wait_for_binary_
             wait_for_binary_proto = self.force_wait_for_cluster_start
         if wait_other_notice is None:
             wait_other_notice = self.force_wait_for_cluster_start
-        self.debug(f"start_nodes: no_wait={no_wait} wait_for_binary_proto={wait_for_binary_proto} wait_other_notice={wait_other_notice} force_wait_for_cluster_start={self.force_wait_for_cluster_start}")
-        self.started=True
+        self.debug(f"start_nodes: no_wait={no_wait} wait_for_binary_proto={wait_for_binary_proto} wait_other_notice={wait_other_notice} force_wait_for_cluster_start={self.force_wait_for_cluster_start} scylla_mode={self.scylla_mode} default_wait_other_notice_timeout={self.default_wait_other_notice_timeout} default_wait_for_binary_proto={self.default_wait_for_binary_proto}")
```
Comment (fruch) [inline ccmlib/scylla_cluster.py:101]: don't we have enough prints showing the modes ?

why do we need it on each node starting ?

File: ccmlib/scylla_cluster.py [hunk 2]
```diff
@@ -98,8 +98,7 @@ def start_nodes(self, nodes=None, no_wait=False, verbose=False, wait_for_binary_
             wait_for_binary_proto = self.force_wait_for_cluster_start
         if wait_other_notice is None:
             wait_other_notice = self.force_wait_for_cluster_start
-        self.debug(f"start_nodes: no_wait={no_wait} wait_for_binary_proto={wait_for_binary_proto} wait_other_notice={wait_other_notice} force_wait_for_cluster_start={self.force_wait_for_cluster_start}")
-        self.started=True
```
Comment (fruch) [inline ccmlib/scylla_cluster.py:102]: why did you remove this ?

**Review by fruch — CHANGES_REQUESTED** [review]
return "self.started=True" back

File: ccmlib/scylla_cluster.py (hunk 2 above)
Comment (bhalevy) [reply ccmlib/scylla_cluster.py:102]: this was inadvertent

**Comment by bhalevy** [comment]
> return "self.started=True" back

done

**Comment by bhalevy** [comment]
BTW setting self.started=True seems to be happening too early.
I'd consider doing that only before the function returns successfully

File: ccmlib/scylla_cluster.py (hunk 1 above)
Comment (bhalevy) [reply ccmlib/scylla_cluster.py:101]: We have a printout in dtest_setup.py `init_default_config` and or printed the apparently correct mode:
```
06:51:00,723 790     dtest_setup                    DEBUG    dtest_setup.py      :694  | test_cluster_expansion_with_cdc[Single_cluster]: Scylla mode is 'debug'
```

But timeout==420 is an evidence for the mode not set up properly in `ScyllaCluster.__init__`

Comment (bhalevy) [reply ccmlib/scylla_cluster.py:101]: I can move the printout to the constructor so it's printed once per cluster instantiation.

Comment (bhalevy) [reply ccmlib/scylla_cluster.py:101]: or print the extra information only `if not self.started`

**Comment by bhalevy** [comment]
Besides fixing this regex, I think that the issue with scylla_mode in `ScyllaCluster.__init__` could be this:
https://github.com/scylladb/scylla-ccm/blob/78b5d266dca3daa209406ab14265fe61479c51e5/ccmlib/scylla_cluster.py#L33-L35

While we pass a valid `cassandra_version` in this case:
https://github.com/scylladb/scylla-dtest/blob/a9b0676fc11eeab144d6d8d5fd402b4db6505dee/dtest_setup.py#L805-L809

**Comment by bhalevy** [comment]
@fruch please re-review

**Comment by bhalevy** [comment]
https://github.com/scylladb/scylla-ccm/commit/168057e1e344f7648b34930e87393e834d191112 also fixes https://github.com/scylladb/scylla-ccm/pull/516#issuecomment-1766222735 by setting the default timeouts again after ScyllaCluster.scylla_mode is correctly extracted in `load_from_repository`

**Review by fruch — APPROVED** [review]
LGTM

**Comment by bhalevy** [comment]
@scylladb/ccm-maint please merge
